Packs/MicrosoftTeams/Integrations/MicrosoftTeams/MicrosoftTeams.py

```diff
@@ -3136,6 +3224,103 @@ def auth_type_switch_handling():
         set_integration_context(integration_context)
 
 
+def expand_permission_list(permissions: list[str]) -> set[str]:
+    """
+    Takes the given permission list and extends it to include relevant lower privileged sub permissions
+    that are supported by the given permissions. (e.g. if theres Chat.ReadWrite then Chat.Read is implicitly supported)
+    :param permissions: Permissions list extracted from the cached graph token
+
+    :return: Extended permissions list including lower privileged permissions supported by the original permissions.
+    """
+    expanded_permissions = set()
+    for permission in permissions:
+        expanded_permissions.add(permission)
+        if permission in HIGHER_PERMISSIONS:
+            expanded_permissions.update(HIGHER_PERMISSIONS[permission])
+
+    demisto.debug(f'Expanded token permission list: {expanded_permissions}')
+    return expanded_permissions
+
+
+def create_missing_permissions_section(missing_permissions: list[str], permission_type: str) -> str:
+    """
+    Generates a detailed error message for the missing permissions.
+
+    :param missing_permissions: List of missing permissions
+    :param permission_type: The type of permissions ('Application' or 'Delegated')
+    :return: Error message string
+    """
+    error_message = f"""\
+The current access token is missing the following permissions:
+{permission_type} - {", ".join(missing_permissions)}
+
+To resolve the issue, add the missing permissions to your bot in the Azure portal:
+Click API permissions > Add a permission > Microsoft Graph > {permission_type} permissions, add the required missing permissions \
+and grant admin consent.
+
+Once done, reset your token using the "!microsoft-teams-auth-reset" command.
+"""
+
+    if AUTH_TYPE == AUTHORIZATION_CODE_FLOW:
+        error_message += (
+            'Then, generate a new auth code using "!microsoft-teams-generate-login-url".\n'
+        )
+
+    error_message += (
+        'The "!microsoft-teams-token-permissions-list" command can be used to '
+        'list the permissions of the currently used graph access token.'
+    )
+
+    return error_message
+
+
+def insufficient_permissions_error_handler() -> str:
+    """
+    Retrieve the known permission requirements for the failed command and give a detailed error message
+    indicating the missing permissions and how to resolve the issue.
+
+    :return: Error message string with instructions regarding the missing permissions
+    """
+    command: str = demisto.command()
+    error_message = []
+    demisto.debug(f"Authentication type is: {AUTH_TYPE}")
+    missing_permissions = []
+    required_permissions_for_command: list = COMMANDS_REQUIRED_PERMISSIONS.get(command, [])
+    permission_type = 'Application' if AUTH_TYPE == CLIENT_CREDENTIALS_FLOW else 'Delegated'
+    if not required_permissions_for_command:
```

Comment:
As i understand here the code checks if a given command is present in the commands required permission list way is the error related to Microsoft?